src/resources/filters/quarto-pre/code-filename.lua

```diff
@@ -10,6 +10,9 @@
 
 function code_filename()
   local function codeBlockWithFilename(el, filename)
+    if quarto.doc.is_format("latex") then
+      filename = string.gsub(filename, "_", "\\_")
+    end
```

Comment:
I think format specific stuff for DecoratedCodeBlock are best put inside the renderer itself. 
I would put this inside https://github.com/quarto-dev/quarto-cli/blob/7dab87bbabd8f8193ae4ff58b9891c6f015d0c49/src/resources/filters/customnodes/decoratedcodeblock.lua#L66-L70

Also I don't know if there could be other escaping needed, but we have a `stringEscape` we use for caption in LaTeX already. 
https://github.com/quarto-dev/quarto-cli/blob/7dab87bbabd8f8193ae4ff58b9891c6f015d0c49/src/resources/filters/quarto-pre/table-captions.lua#L223C1-L225

This would cover `_` and also other required escaping probably

Reply:
Using `escapeString` works properly 6eb06fb41c421e5f1a418c49af871f4942fc5a22